### PR TITLE
 migration to 2022 compiler

### PR DESCRIPTION
**Pull request by makart19**
made required changes to enable build with latest 2022.2.22 compiler

### CODE REVIEW

**Comment by makart19** [comment]
/intelci: run
set infra_branch=dev/amk_migr_to_new_2022_cmp

File: cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp [hunk 1]
```diff
@@ -648,7 +648,7 @@ compute_kernel_dense_impl<Float, List>::merge_blocks(local_buffer_list<Float, Li
         cgh.parallel_for(nd_range, [=](sycl::nd_item<1> item) {
             const std::int64_t local_size = item.get_local_range()[0];
             const std::int64_t id = item.get_local_id()[0];
-            const std::int64_t group_id = item.get_group().get_id(0);
+            const std::int64_t group_id = item.get_group().get_group_id(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp:651]: You can do that easier:
```suggestion
            const std::int64_t group_id = item.get_group(0);
```

That [implies the function from specification](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#nditem-class):
```
template <int dim>
class nd_item {
...
size_t get_group(int dimension) const;
...
};
```

**Review by KulikovNikita — APPROVED** [review]
Approved. But please consider to use `.get_group(int dim)` directly.

**Pull request merged**